rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -249,14 +249,47 @@ def _maybe_enrich_balancer_v1_events(
                     event.event_type == HistoryEventType.SPEND and
                     event.event_subtype == HistoryEventSubType.NONE
                 ):
-                    event.event_type = HistoryEventType.STAKING
+                    event.event_type = HistoryEventType.WITHDRAWAL
                     event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
                     event.counterparty = CPT_BALANCER_V1
                     event.notes = f'Return {event.balance.amount} {token.symbol} to a Balancer v1 pool'  # noqa: E501
                     return True
 
         return False
 
+    def _check_refunds_v1(
+            self,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> list[HistoryBaseEntry]:
+        """
+        It can happen that after sending tokens to the DSProxy in balancer V1 the amount of tokens
+        required for the deposit is lower than the amount sent and then those tokens are returned
+        to the DSProxy and then to the user.
+        """
+        deposited_assets = []
+
+        for event in decoded_events:
+            if (
+                event.counterparty == CPT_BALANCER_V1 and
+                event.event_type == HistoryEventType.DEPOSIT and
+                event.event_subtype == HistoryEventSubType.DEPOSIT_ASSET
+            ):
+                deposited_assets.append(event.asset)
+            elif (
+                event.counterparty == CPT_BALANCER_V1 and
+                event.event_type == HistoryEventType.DEPOSIT and
+                event.event_subtype == HistoryEventSubType.RECEIVE_WRAPPED and
+                event.asset in deposited_assets
+            ):
+                # in this case we got refunded one of the assets deposited
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
```

Comment:
I'm not so sure about using `REMOVE_ASSET` here. 
Also, shouldn't you change the `event_type` from `DEPOSIT` to `RECEIVE`?

Reply:
It's confusing, I agree. You guys seem to pick new types/subtypes combinations for everything depending on how you feel on a given day. This is confusing as heck and does not scale. Will bite us in the ass and does not let us generalize for common types of events between protocols.

We don't have a refund type though so not sure what the right approach here is either.

We need to generalize all the combinations and write rules for them.

Reply:
As discussed in discord the types here follow the curve ones now and I'll create a list + enforce the types in the decoders